client/src/cordova/dev.action.mjs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import os from 'os';
+import path from 'path';
+import url from 'url';
+
+import {createReloadServer} from '@outline/infrastructure/build/create_reload_server.mjs';
+import {getRootDir} from '@outline/infrastructure/build/get_root_dir.mjs';
+import {runAction} from '@outline/infrastructure/build/run_action.mjs';
+import {spawnStream} from '@outline/infrastructure/build/spawn_stream.mjs';
+import chalk from 'chalk';
+import {hashElement} from 'folder-hash';
+import * as fs from 'fs-extra';
+
+import {makeReplacements} from '../../build/make_replacements.mjs';
+
+const OUTPUT_PATH = 'output/build/client/macos';
+const OUTLINE_APP_PATH = 'Debug/Outline.app';
+const OUTLINE_APP_WWW_PATH = 'Contents/Resources/www';
+const RELOAD_SERVER_PORT = 35729;
+
+const getUIHash = async () => {
+  const hashResult = await hashElement(
+    path.join(getRootDir(), 'client/src/www'),
+    {
+      files: {include: ['**/*.ts', '**/*.html', '**/*.css', '**/*.js']},
+    }
+  );
+
+  return hashResult.hash;
+};
+
+/**
+ * @description Builds the parameterized cordova binary (ios, macos, maccatalyst, android).
+ *
+ * @param {string[]} parameters
+ */
+export async function main() {
+  console.warn(
+    chalk.yellow(
+      'This action only works for the MacOS platform. Ignoring all inputs.'
```

Comment:
Why not parse the inputs and exit early or throw an error if it's not targeting macos? It would be more explicit.